### ISSUE TITLE
Webview - visible/invisible 

### ISSUE THREAD

**Issue by dani2003** [myflashlab/webView-ANE#55]
You can make visible and invisible ... Webview
Es.
...
_ex = new RichWebView
_ex.visible = false
_ex.visible = true
...

Thank you

**Comment by myflashlab** [comment]
Thanks for your input. I just marked this ticket and we will consider it in the next release.

**Comment by dani2003** [comment]
Thanks for your help, very nice!

There are two big problems: 
1) RichWebViewEvent.PAGE_FINISHED is called multiple times. Very strange.!
2) AirBridge.evoke () it is never called if put in the <body onload = 'AirBridge.evoke ("myJsonString")'>;
I have to be sure that the page loads to populate it dynamically from Flash.
How can I fix without using setTimeout in javascritps.

**Comment by myflashlab** [comment]
One thing to remember is that the AirBridge JS is injected into the html page only after the page is fully loaded. so if you are calling the evoke method prior to that, it won't work simply because the JS is not available yet.

**Comment by PapaDeBeau** [comment]
Yes, I too would love to know an efficient way to call the AirBridge once the page has fully loaded. I have tried to have it run on the PageFinished but I got errors and abandoned it. It would be great to know how to call it when the dom and all js is totally loaded.

**Comment by myflashlab** [comment]
@galaxybeing @PapaDeBeau  @dani2003 we will put some investigation on this to see if we can effectively inject the ```AirBridge.js``` sooner than waiting for the page being loaded fully. as for now, keep in mind that the injection happens only after ```_ex.addEventListener(RichWebViewEvent.PAGE_FINISHED, onPageFinished);``` is dispatched which means the page is fully loaded.

maybe you should call a function from Air in your JS as soon as the page is loaded to tell JS that injection is now completed? just some thoughts.

**Comment by galaxybeing** [comment]
First off, I’m not checking if the injection happens until I click a button, so plenty of time has passed. If it takes that long it’s not really useful.   I’m not going to throw up a dialog that says “wait before you click on any buttons.”

 

And second, Bridge.call works!  That solves our problem. We have already launched our iOS app with the old ANE and using Bridge code.  If Bridge work as well, that would really help us in this transition phase because we don’t have to modify our website for the new ANE used in the Android version.

 

Our problem seems to be just that we have poor checking of whether it is mobile or not (works under ios, but not under Chrome on Android).  Once we fix that, if the Bridge code still works we can release the Android version with the new ANE with the video iframe fixed.

 

Maybe you should just keep the Bridge code permanently.  If the injection is unreliable.

 

I did have one question about  onReceivedMassage.  Does it include a happy ending?  ;)

**Comment by myflashlab** [comment]
Hmm, for us to be able to solve this problem the best way possible, please, would you confirm the following:

run your app in debug mode, and listen for ```_ex.addEventListener(RichWebViewEvent.PAGE_FINISHED, onPageFinished);``` in your flash side. only after that listener tells you that your page load is completed, try to call ```AirBridge.evoke``` from JS. Let me know if that works then? and it's not working prior to the PAGE_FINISHED listener?

the ```Bridge.call``` works because it is already in JS without the injection thing.

We appreciate your inputs and comments on this. we will be gathering all the required information and our dev team will put an urgent upgrade on this after we are sure we have correctly identified the best possible solution.

**Comment by galaxybeing** [comment]
I just realized my problem was that I didn't have the AirBridge.js file in the directories.  But I am finewith just using Bridge.call for now since that's all implemented and working.

**Comment by myflashlab** [comment]
ok, good. just one suggestion I should make here is that it would be a good idea if you update your JS code from ```Bridge.call``` to ```AirBridge.evoke``` because in all future versions we will be using the new setup and so it would be easier for you to keep your app updated.

although I'll be closing this ticket now, as it's namely related to making the webview in/visible, we have the features marked down in our to-do list and will apply them with the next releases.

**Comment by galaxybeing** [comment]
Sorry to reopen this, but I just checked my 5.1 release directory and there is no AirBridge.js file in it.  I don't want to create a new ticket.  Should I?

I found it on the site of course and finally looked into it.  Its is basically the same code as Bridge, and the same methodology. We just pasted the Bridge.js code into our pages since it was so small why have another download.  In one case we have had to modify it because it was not detecting ios soon enough for our purposes and using the android method on an ios device.
 (We needed to send to the app right after document ready.  With injection, you probably don't even have any AirBridge code available there)
 
So yeah, there's always going to be timing issues right at startup.  The injection is a great idea.  But with some pages it might be better just to paste the Bridge code in so it's there when you need it at document ready.  We also renamed the function to Bridge.callFlash();  

Bla bla bla ok I said my peace.  you can close it now. :)

**Comment by myflashlab** [comment]
the AirBridge.js is here https://github.com/myflashlab/webView-ANE/tree/master/FD/bin
what is important to be remembered is that it must be at the root of the assets folder which means, just next to the main .swf file.

yes @galaxybeing your findings are totally correct. The only reason we decided to go with the injection idea was that many devs had problem with using the js file manually. But as you said, the methodology is the same and it's a very simple JS file anyway.

anyhow, we are working on this to make things less annoying. we do appreciate your kind feedback. These comments will surely help us improve our ANEs.

**Comment by myflashlab** [comment]
visible property is now added to V6.0.0
